stan/math/prim/scal/meta/scalar_seq_view.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_SCALAR_SEQ_VIEW_HPP
+#define STAN_MATH_PRIM_SCAL_META_SCALAR_SEQ_VIEW_HPP
+
+#include <stan/math/prim/scal/meta/scalar_type.hpp>
+
+namespace stan {
+
+  template <typename C, typename T = typename scalar_type<C>::type>
+  class scalar_seq_view {
+  public:
+    explicit scalar_seq_view(const C& c) : c_(c) {}
+
+    /**
+     * Segfaults if out of bounds.
+     * @param i index
+     * @return the element at the specified position in the container
+     */
+    const T& operator[](int i) const {
```

Comment:
What do you guys think about these `operator[]`s being inlined? I'm still getting a feel for where that's appropriate.